### PR TITLE
[Vectorize] Changes to support vectorize feature

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Implement vectorize

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/NVidiaEmbeddingClient.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.List;
+import java.util.Optional;
+import org.eclipse.microprofile.rest.client.annotation.ClientHeaderParam;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+/**
+ * Interface that accepts a list of texts that needs to be vectorized and returns embeddings based
+ * of chosen nvidia model.
+ */
+public class NVidiaEmbeddingClient implements EmbeddingProvider {
```
Comment (vkarpov15) [inline src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/NVidiaEmbeddingClient.java:19]: Minor nitpick but the company name is "Nvidia", all one word, so the uppercase "V" is incorrect AFAIK

File: src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/OpenAiEmbeddingClient.java [hunk 2]
```diff
@@ -1,37 +1,39 @@
...
 import jakarta.ws.rs.Path;
 import java.net.URI;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 import org.eclipse.microprofile.rest.client.annotation.ClientHeaderParam;
 import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
 
-public class OpenAiEmbeddingClient implements EmbeddingService {
+public class OpenAiEmbeddingClient implements EmbeddingProvider {
```
Comment (vkarpov15) [inline src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/OpenAiEmbeddingClient.java:15]: Similar minor nitpick, the company name is "OpenAI" with an uppercase "I", so I'd prefer the EmbeddingClient name to match the company name exactly unless there's a good reason not to. And you already have "VertexAI" so there's precedent for uppercase "I"

**Review by vkarpov15 — COMMENTED** [review]
LGTM so far, just a couple of pedantic comments

File: src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/NVidiaEmbeddingClient.java (hunk 1 above)
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/NVidiaEmbeddingClient.java:19]: Will create a separate to address all provider names.

**Review by jeffreyscarpenter — APPROVED** [review]

**Pull request merged**